src/backup.py

```diff
@@ -3,18 +3,47 @@
 
 """Provides backup functionality for Synapse."""
 
+import datetime
 import logging
-from typing import Any, Optional
+import os
+from typing import Any, Dict, Iterable, List, Optional
 
 import boto3
+import ops
 from botocore.config import Config
 from botocore.exceptions import BotoCoreError, ClientError
 from botocore.exceptions import ConnectionError as BotoConnectionError
+from ops.pebble import APIError, ExecError
 from pydantic import BaseModel, Field, validator
 
+import synapse
+
+AWS_COMMAND = "/aws/dist/aws"
+
+# The configuration files to back up consist in the signing keys
+# plus the sqlite db if it exists.
+BACKUP_FILE_PATTERNS = ["*.key", "homeserver.db*"]
+
+# For the data directory, inside the "media" directory, all directories starting
+# with local_ will be backed up. The directories starting with "remote_" are from
+# other server and is it not necessary to back them up.
+MEDIA_DIR = "media"
+LOCAL_DIR_PATTERN = "local_*"
+
+# A smaller value will minimise memory requirements. A bigger value can make the transfer faster.
+S3_MAX_CONCURRENT_REQUESTS = 1
+PASSPHRASE_FILE = "/root/.gpg_passphrase"  # nosec
```

Comment:
Not sure this is the right place. Any idea?

Reply:
Do you mean the key location? This would imply running synapse as root user, which contravenes the least privilege principle

Reply:
Perhaps under the application directory (`/srv/synapse/...` I guess) or under `$HOME` (wherever that is)?

Reply:
I have updated the location of the passphrase file to Synapse config dir. I haven't changed to run it as the Synapse user as it is managed by the startup script and it is not fully configured (has no home dir for example).

WDYT?